### PR TITLE
Add option to authenticate using userserv

### LINKED ISSUE #1237: Chanserv auth_method?
Is there an option for the [auth_method](https://sopel.chat/docs/config.html#the-core-configuration-section) that's equivalent to `chanserv`?

### PR DESCRIPTION
**Pull request by jpaulovic**
@dgw edit: Closes #1237

### CODE REVIEW

File: sopel/coretasks.py [hunk 1]
```diff
@@ -59,6 +59,12 @@ def auth_after_register(bot):
             account + ' ' + password
         ))
 
+    elif bot.config.core.auth_method == 'userserv':
+	account = bot.config.core.auth_username
+	password = bot.config.core.auth_password
+	bot.msg("USERSERV", "LOGIN {account} {password}".format(
+		account=account, password=password))
+
```
Comment (dgw) [inline sopel/coretasks.py]: This block adds a `TabError` because some of the lines use tabs instead of spaces for indentation.

Comment (jpaulovic) [reply sopel/coretasks.py]: fixed

File: sopel/coretasks.py [hunk 2]
```diff
@@ -59,6 +59,12 @@ def auth_after_register(bot):
             account + ' ' + password
         ))
 
+    elif bot.config.core.auth_method == 'userserv':
+        account = bot.config.core.auth_username
+        password = bot.config.core.auth_password
+        bot.msg("USERSERV", "LOGIN {account} {password}".format(
```
Comment (dgw) [inline sopel/coretasks.py]: The UserServ name shouldn't be hardcoded. See #1238, which uses `bot.config.core.auth_target` if configured.

**Comment by dgw** [comment]
@paoloantinori Sure, why not.

**Comment by paoloantinori** [comment]
Thank you!

**Pull request merged**